File: 백준/Silver/1269. 대칭 차집합/대칭 차집합.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n =  Integer.parseInt(st.nextToken());
        int m =  Integer.parseInt(st.nextToken());

        Set<Integer> a = new HashSet<>();
        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < n; i++){
            a.add(Integer.parseInt(st.nextToken()));
        }

        Set<Integer> b = new HashSet<>();
        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < m; i++){
            b.add(Integer.parseInt(st.nextToken()));
        }

        Set<Integer> intersection = new HashSet<>(a);
        intersection.retainAll(b);

        a.removeAll(intersection);
        b.removeAll(intersection);

        Set<Integer> union = new HashSet<>(a);
        union.addAll(b);

        StringBuilder sb = new StringBuilder();
        sb.append(union.size());

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/1269. 대칭 차집합/대칭 차집합.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    var st = StringTokenizer(br.readLine())

    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()

    val a = mutableSetOf<Int>()
    st = StringTokenizer(br.readLine())
    repeat(n) {
        a.add(st.nextToken().toInt())
    }

    val b = mutableSetOf<Int>()
    st = StringTokenizer(br.readLine())
    repeat(m) {
        b.add(st.nextToken().toInt())
    }

    val inter = a.intersect(b) as MutableSet<Int>
    val result = a.size + b.size - 2 * inter.size

    val sb = StringBuilder()
    sb.append(result)

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}